### PR TITLE
[COEUS-55] Create API service for the Algorithm

### PR DESCRIPTION
**Pull request by keiseithunder**

### CODE REVIEW

File: Client.py [hunk 1]
```diff
@@ -226,19 +227,19 @@ def algorithm(
                     _this_paragraph_dictionary[noun][answer] = dict()
             return
         else:
-            data = [sent.text for sent in sentences]
+            data = [sent.text for sent in number_of_sentences]
```
Comment (Pittawat2542) [inline Client.py]: Should it be `range(number_of_sentences)`?

File: Client.py [hunk 2]
```diff
@@ -226,19 +227,19 @@ def algorithm(
...
             def distance(x, y):
-                return Client.clustering_model.wmd.wmdistance(Client.input_preparator.preprocess(
+                return Client.wmd.wmdistance(Client.input_preparator.preprocess(
                     data[int(x[0])]), Client.input_preparator.preprocess(data[int(y[0])]))
 
             proximity_matrix = pairwise_distances(X, X, metric=distance)
-            best_k, best_cluster = Client.clustering_model.predict(
-                X, len(sentences), proximity_matrix)
+            best_k, best_cluster = Client.clustering_repository.getData(
+                X, len(number_of_sentences), proximity_matrix.tolist())
```
Comment (Pittawat2542) [inline Client.py]: `number_of_sentences` is already number, so no need to wrap in `len()` again?

File: Client.py [hunk 3]
```diff
@@ -257,7 +258,7 @@ def distance(x, y):
                         _this_paragraph_dictionary[noun][answer] = dict()
             return Client.algorithm(
                 original_text,
-                sentences.join('. '),
+                number_of_sentences.join('. '),
```
Comment (Pittawat2542) [inline Client.py]: 🤔

File: app.py [hunk 4]
```diff
@@ -1,21 +1,57 @@
...
+            token = request.headers.get('Authorization').split(' ')[1]
+            if token != authentication_token:
+                return jsonify({"error": "Unauthorized request"}), 403
+        wiki_path = request.json['wiki_path']
+        if wiki_path is not None:
+            try:
+                result = Client.generate_mind_map_from_semi_structure_text(topic)
+                return jsonify({"result": result})
+            except Exception as e:
+                print(wiki_path)
```
Comment (Pittawat2542) [inline app.py]: Should you add `, flush=True` ?

File: app.py [hunk 5]
```diff
@@ -1,21 +1,57 @@
...
+                print(e)
```
Comment (Pittawat2542) [inline app.py]: Should you add `, flush=True` ?

File: app.py [hunk 6]
```diff
@@ -1,21 +1,57 @@
...
+            if token != authentication_token:
+                return jsonify({"error": "Unauthorized request"}), 403
+        topic = request.json['topic']
+        text = request.json['text']
+        if topic is not None and text is not None:
+            try:
+                result = Client.generate_mind_map_from_unstructured_text(topic, text)
+                return jsonify({"result": result})
+            except Exception as e:
+                print(topic)
```
Comment (Pittawat2542) [inline app.py]: Should you add `, flush=True` ?

File: app.py [hunk 7]
```diff
@@ -1,21 +1,57 @@
...
+                print(e)
```
Comment (Pittawat2542) [inline app.py]: Should you add `, flush=True` ?

**Review by Pittawat2542 — CHANGES_REQUESTED** [review]

File: Client.py (hunk 1 above)
Comment (keiseithunder) [reply Client.py]: this is to get sentence fixed after spacy change api

**Review by Pittawat2542 — APPROVED** [review]

**Pull request merged**